### PR TITLE
[front] feat: display entity contexts on comparison page

### PR DESCRIPTION
**Pull request by GresilleSiffle**
**related issues** #1852

---

### Description

This PR displays the entity contexts in the comparison interface.

The boxes can be collapsed, and this state is saved per entity in the browser local storage.

On small screens and mobile, the two boxes stack on top of each other. 

**to-do**
- [x] do not display two context box when both selectors display the same entity
- [x] make the content box collapsible
- [x] save the collapsed state of contexts in the local storage
- [x] when two context are displayed, make it clear that context A is related to video A, and context B to video B
- [x] reload the context text when the UI language changes

### Preview

The title of the entity is also displayed before the context to connect more explicitly the entity with its context.

capture

The following captures have been made before the introduction of the entity title.

| one entity with context | two entities with context | two folded contexts |
|---|---|---|
| capture1 | capture2 | capture3 |

| small screens |
|---|
| capture4 |

### Checklist

- [x] I added the related issue(s) id in the related issues section (if any)
  - if not, delete the related issues section
- [x] I described my changes and my decisions in the PR description
- [x] I read the development guidelines of the [CONTRIBUTING.md][development-guidelines]
- [x] The tests pass and have been updated if relevant
- [x] The code quality check pass

[development-guidelines]: https://github.com/tournesol-app/tournesol/blob/main/CONTRIBUTING.md#development-guidelines

### CODE REVIEW

File: frontend/src/features/comparisons/Comparison.tsx [hunk 1]
```diff
@@ -137,13 +148,34 @@ const Comparison = ({
...
+          initializing.current = false;
         });
   }, [pollName, uidA, uidB, selectorA.uid, selectorB.uid]);
 
+  /**
+   * When the UI language changes, refresh the ratings to retrieve the
+   * corresponding localized entity contexts.
+   */
+  useEffect(() => {
+    if (initializing.current || isLoading) {
```
Comment (amatissart) [inline frontend/src/features/comparisons/Comparison.tsx:165]: I don't understand why this check on `initliazing.current` is useful here :thinking:

File: frontend/src/features/entity_context/EntityContextBox.tsx [hunk 2]
```diff
@@ -96,25 +124,58 @@ const EntityContextList = ({
...
+            <strong>
+              {origin_ === OriginEnum.ASSOCIATION &&
+                t('entityContext.theAssociationWouldLikeToGiveYouContext')}
+            </strong>
+          )}
+        </AlertTitle>
+
+        {collapsible && (
+          <IconButton
+            aria-label="Show question's answer"
```
Comment (amatissart) [inline frontend/src/features/entity_context/EntityContextBox.tsx:146]: wrong label
```suggestion
            aria-label="Show context"
```

**Review by amatissart — APPROVED** [review]
LGTM with minor comments

File: frontend/src/features/comparisons/Comparison.tsx (hunk 1 above)
Comment (GresilleSiffle) [reply frontend/src/features/comparisons/Comparison.tsx:165]: I checked and it's not useful, the `isLoading` check seems to be enough.

**Pull request merged**